src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -31,28 +31,48 @@ internal sealed class DashboardLifecycleHook(IConfiguration configuration,
     private readonly CancellationTokenSource _shutdownCts = new();
     private Task? _dashboardLogsTask;
 
-    public Task BeforeStartAsync(DistributedApplicationModel model, CancellationToken cancellationToken)
+    public Task BeforeStartAsync(DistributedApplicationModel appModel, CancellationToken cancellationToken)
     {
         Debug.Assert(executionContext.IsRunMode, "Dashboard resource should only be added in run mode");
 
-        if (model.Resources.SingleOrDefault(r => StringComparers.ResourceName.Equals(r.Name, KnownResourceNames.AspireDashboard)) is { } dashboardResource)
+        if (appModel.Resources.SingleOrDefault(r => StringComparers.ResourceName.Equals(r.Name, KnownResourceNames.AspireDashboard)) is { } dashboardResource)
         {
             ConfigureAspireDashboardResource(dashboardResource);
 
             // Make the dashboard first in the list so it starts as fast as possible.
-            model.Resources.Remove(dashboardResource);
-            model.Resources.Insert(0, dashboardResource);
+            appModel.Resources.Remove(dashboardResource);
+            appModel.Resources.Insert(0, dashboardResource);
         }
         else
         {
-            AddDashboardResource(model);
+            AddDashboardResource(appModel);
         }
 
         _dashboardLogsTask = WatchDashboardLogsAsync(_shutdownCts.Token);
 
         return Task.CompletedTask;
     }
 
+    public Task AfterResourcesCreatedAsync(DistributedApplicationModel appModel, CancellationToken cancellationToken = default)
+    {
+        if (appModel.Resources.SingleOrDefault(r => StringComparers.ResourceName.Equals(r.Name, KnownResourceNames.AspireDashboard)) is { } dashboardResource)
```

Comment:
#5750 adds the concept of a "system" resource annotation, which could be used to filter here in a way that doesn't have special knowledge about the dashboard. We will also want to filter out stop/restart for dashboard extensions here too I suspect, which are also identifiable by annotation.

Reply:
A systems resource to where? The resource server? Or the app model?

Reply:
Both. It's an annotation on the resource in the app model, and exposed to clients of the resource service in the proto.